src/main/java/com/employeesService/EmployeesService/repository/JpaDepartmentRepository.java

```diff
@@ -0,0 +1,72 @@
+package com.employeesService.EmployeesService.repository;
+
+import com.employeesService.EmployeesService.model.Department;
+import com.employeesService.EmployeesService.model.DepartmentEmployee;
+import com.employeesService.EmployeesService.model.DepartmentEmployeeId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class JpaDepartmentRepository implements DepartmentRepository {
+
+    private final EntityManager entityManager;
+
+    @Autowired
+    public JpaDepartmentRepository(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    @Override
+    @Transactional
+    public Department save(Department department) {
+        this.entityManager.persist(department);
+        return department;
+    }
+
+    @Override
+    public List<Department> findAll() {
+        return this.entityManager.createQuery("SELECT d FROM Department d ", Department.class).getResultList();
+    }
+
+    @Override
+    public Optional<Department> findById(String deptNo) {
+        try {
+            return Optional.ofNullable(this.entityManager.find(Department.class, deptNo));
+        } catch (EmptyResultDataAccessException e) {
```

Comment:
I'm saying for many times now, please deal with this exception, prove to me it's used, otherwise REMOVE it!!!